fraud/proof.go

```diff
@@ -80,3 +93,40 @@ func Unmarshal(proofType ProofType, msg []byte) (Proof, error) {
 	}
 	return unmarshaler(msg)
 }
+
+// MonitorProofs enables metrics to monitor fraud proofs.
+func MonitorProofs(proofType ProofType) func(datastore.Datastore) {
+	return func(ds datastore.Datastore) {
+		store := initStore(proofType, ds)
+		counter, _ := meter.AsyncInt64().Counter(string(proofType),
+			instrument.WithUnit(unit.Dimensionless),
+			instrument.WithDescription("Stored fraud proof"),
+		)
+		err := meter.RegisterCallback(
+			[]instrument.Asynchronous{
+				counter,
+			},
+			func(ctx context.Context) {
+				proofs, err := getAll(ctx, store, proofType)
```

Comment:
Can we make it over Service API am not over private methods?

Reply:
Sounds like it can actually be combined with #1040 then, no? Since that will add metering to Service

Reply:
Or do you mean the getAll specifically?